sdk/storage/azure-storage-perf/pom.xml

```diff
@@ -45,6 +45,24 @@
       <artifactId>perf-test-core</artifactId>
       <version>1.0.0-beta.1</version> <!-- {x-version-update;com.azure:perf-test-core;current} -->
     </dependency>
+
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-http-netty</artifactId>
+      <version>1.13.0-beta.1</version> <!-- {x-version-update;com.azure:azure-core-http-netty;current} -->
```

Comment:
@mikeharder all of these particular dependencies, azure-core-http-netty, azure-core-http-okhttp and azure-core should be using the dependency version not the current version. Any library built in another service directory should only ever be using the dependency version. perf-test-core, being in /common, is an exception and is correctly using the current version.

Reply:
This is a perf test, not a "library" in the typical sense.  We do want perf tests to use the current version (the sources in main) by default.  The same we would do for unit tests or functional tests.

Reply:
The "java - storage - ci" pipeline did have a lot of failures for this PR, but they seem unrelated to my changes, unless it's somehow related to "dependency version vs current version".  This was the error:

```
2022-06-17T07:53:07.2059193Z [ERROR] codesnippet-maven-plugin has encountered errors while updating codesnippets.
```

Reply:
@kasobol-msft, @alzimmermsft, @jaschrep-msft: Do you agree the storage perf tests (and other library perf tests) should reference the "current" version of all internal dependencies?  I think this is the correct design from the perf perspective.

Reply:
I think these dependencies might be breaking the build somehow.  I know `azure-storage-perf` builds fine with these dependencies on Java 8, because we do it in the perf pipelines.  And I believe `azure-storage-perf` builds fine without these dependencies on Java 17, because it's building fine in main.  But it appears the combination of these dependencies and Java 17 fail to build.

I'll try temporarily removing these dependencies to see if it makes the builds for this PR green.  And I'll also try using Java 17 in the perf pipeline to see if it fails.

Reply:
Yes, this is correct choice given current state of art.

The other option is to change what "source" version means in perf pipeline. 
Today it means "use whatever is in pom.xml".
It could mean "go to version_client.txt" pull "current" and replace it with that.

However, I suggest to take this change as is for now and create work item if follow up is needed. We shouldn't block perf testing on this, not shipping regressions is more important than being strict in perf packages.

Reply:
@kasobol-msft: I think these might be breaking the build somehow.  I don't understand the build errors but I don't think anything else in this PR would cause them.

Could you or someone from the Java team take a look at the errors in this build?

https://dev.azure.com/azure-sdk/public/_build/results?buildId=1650053&view=results

Reply:
This build was deleted because I modified the PR.  I'm trying a new build without these dependencies to see if it passes.  Then I can restore the dependencies and we can investigate the build failures.

Reply:
> The other option is to change what "source" version means in perf pipeline.
> Today it means "use whatever is in pom.xml".
> It could mean "go to version_client.txt" pull "current" and replace it with that.

Yes, the perf automation could upgrade pom.xml to "current".  However, we should also consider the scenario of a dev running the perf test manually on their dev machine.  In this scenario, I think we want to use the "current" version of all deps.

Reply:
To unblock this PR, I have reverted the changes to `azure-storage-perf/pom.xml`.  For now, the perf pipeline will measure storage from sources and core from packages.  I will see if I can adjust the perf automation to add the source versions of core packages dynamically at runtime.